server/parsec/components/postgresql/invite.py

```diff
@@ -917,7 +921,12 @@ async def _conduit_listen(
             # has switched to it next state.
 
             if curr_conduit_state == NEXT_CONDUIT_STATE[ctx.state] and curr_our_payload is None:
-                return ctx.peer_payload, is_last_exchange
+                # Careful here: it's possible that peer has already changed the state
+                # of the conduit by the time we reach this point. The means it might
+                # have already changed the `last_exchange` flag. We should not rely on.
+                # Instead, we return the one we captured during the `conduit_talk`,
```

Comment:
```suggestion
                # of the conduit by the time we reach this point. This means it might
                # have already changed the `last_exchange` flag so we should not rely on it.
                # Instead, we return the flag captured during the `conduit_talk`,
```

Reply:
[195d06b](https://github.com/Scille/parsec-cloud/pull/7307/commits/195d06bc98e169d4a6f8583ab5f38c1bd1ef9de6)